test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -360,6 +360,19 @@ TEST_VM(os, jio_snprintf) {
 //  (from multiple calls to os::reserve_memory)
 static address reserve_multiple(int num_stripes, size_t stripe_len) {
   assert(is_aligned(stripe_len, os::vm_allocation_granularity()), "Sanity");
+
+#ifdef __APPLE__
+  // Workaround: try reserving memory with executable=true
+  // to figure out if that's supported on this macOS version
+  bool executable = true;
```

Comment:
Doesn't this local interfere with the same named local at line 388? Seems bad style to have two locals with the same name regardless.

Reply:
I used to have the code in a separate function, where it didn't conflict, and I now think that factoring that check out into a function is actually the way to go here. What do you think?